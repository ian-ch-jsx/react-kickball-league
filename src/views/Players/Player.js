import { useState, useEffect } from 'react';
import { getPlayersById } from '../../services/players';

export default function Player(props) {
  const [player, setPlayer] = useState({ players: [] });

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPlayersById(props.match.params.id);
      setPlayer(data[0]);
    };
    fetchData();
  }, [props.match.params.id]);
  console.log(player);
  return (
    <div>
      <h1>{player.name}</h1>
      <h2>player info</h2>
      <p>{player.position}</p>
      <p></p>
    </div>
  );
}
