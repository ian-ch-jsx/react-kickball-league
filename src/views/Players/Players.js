//render all players list
import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getPlayers } from '../../services/players';

export default function Players() {
  const [players, setPlayers] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPlayers();
      setPlayers(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <h1>loading</h1>;

  return (
    <ul>
      {players.map((player) => (
        <li key={player.id}>
          <Link to={`/players/${player.id}`}>{player.name}</Link>
        </li>
      ))}
    </ul>
  );
}
