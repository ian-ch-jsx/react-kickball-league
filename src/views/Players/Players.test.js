import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Players from './Players';

test('renders list of players', async () => {
  const { container } = render(
    <MemoryRouter>
      <Players />
    </MemoryRouter>
  );
  await screen.findByText('Betty Grey', 'Deb E. Downer');
  expect(container).toMatchSnapshot();
});
