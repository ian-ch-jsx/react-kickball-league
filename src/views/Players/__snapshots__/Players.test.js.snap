// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of players 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Players
    </h1>
    <p>
      <a
        href="/players/1"
      >
        Bennie Jetts
      </a>
    </p>
    <p>
      <a
        href="/players/12"
      >
        Betty Grey
      </a>
    </p>
    <p>
      <a
        href="/players/7"
      >
        Bud E. Guy
      </a>
    </p>
    <p>
      <a
        href="/players/8"
      >
        Buster Rimes
      </a>
    </p>
    <p>
      <a
        href="/players/9"
      >
        Deb E. Downer
      </a>
    </p>
    <p>
      <a
        href="/players/13"
      >
        Dwayne Elizondo Mountain Dew Herbert Camacho
      </a>
    </p>
    <p>
      <a
        href="/players/15"
      >
        Frito P.
      </a>
    </p>
    <p>
      <a
        href="/players/5"
      >
        Gary Coolman
      </a>
    </p>
    <p>
      <a
        href="/players/2"
      >
        Hans Olo
      </a>
    </p>
    <p>
      <a
        href="/players/10"
      >
        Jayne Phonda
      </a>
    </p>
    <p>
      <a
        href="/players/14"
      >
        Joe Bauers
      </a>
    </p>
    <p>
      <a
        href="/players/11"
      >
        Kara Whollot
      </a>
    </p>
    <p>
      <a
        href="/players/3"
      >
        Ron Long
      </a>
    </p>
    <p>
      <a
        href="/players/4"
      >
        Shorty Biggum
      </a>
    </p>
    <p>
      <a
        href="/players/6"
      >
        Terry Cruise
      </a>
    </p>
  </div>
</div>
`;
